logging_loki/handlers.py

```diff
@@ -86,3 +83,38 @@ def emit(self, record: logging.LogRecord):
             self.emitter(record, self.format(record))
         except Exception:
             self.handleError(record)
+
+    def emit_batch(self, records: list[logging.LogRecord]):
+        """Send a batch of log records to Loki."""
+        # noinspection PyBroadException
+        try:
+            self.emitter.emit_batch([(record, self.format(record)) for record in records])
+        except Exception:
+            for record in records:
+                self.handleError(record)
+
+class LokiBatchHandler(MemoryHandler):
+    interval: float # The interval at which batched logs are sent in seconds
+    _last_flush_time: float
+    target: LokiHandler
+
+    def __init__(self, interval: float, capacity: int = LOKI_MAX_BATCH_BUFFER_SIZE, **kwargs):
+        super().__init__(capacity, **kwargs)
+        self.interval = interval
```

Comment:
I see that in `LokiQueueHandler` the `batch_interval` is optional but when it gets propagated to `LokiQueueHandler` we don't consider `interval` as being optional. Maybe I'm missing something. My worries are related to `(time.time() - self._last_flush_time >= self.interval)` condition used in `shouldFlush` fn because we'll get a TypeError if self.interval is None.

Reply:
https://github.com/waylayio/python-logging-loki/blob/c2005319f3c0607bb183fda7c1f0c9f6f4bac745/logging_loki/handlers.py#L26-L29

In the `LokiQueueHandler` constructor there is check wether or not batch_interval is set? This should be sufficient, right?

Reply:
Now that I see this chunky code i'm gonna change it to use a ternary operator

Reply:
Probably too late for this library, but I would (here and elsewhere) only use dependency injection in the`__init__` methods, and not create objects from properties (like the target `handler` in this case).

You can then still have other static builder methods that create the objects and stitches them together.

Reply:
I would love to add dependency injection as well, but since this repo is a fork, I just want to make minimal changes